### ISSUE TITLE
Email compose body not wysywig in certain cases in 7.11.18

### ISSUE THREAD

**Issue by ebogaard** [salesagility/SuiteCRM#8956]
#### Issue
Since upgrading from 7.11.15 to 7.11.18, there is a problem with the email compose screen in certain cases:
1. the body-field (description) isn't wysiwyg
2. functionality like filling with template information isn't working
3. the field is wider than the containing object

You'll experience these issues when clicking on an e-mail address when in DetailView of a Contact.
When you click on the same e-mail address of the same Contact in ListView, all works at it should.

I cleared the browser cache while testing this.
Reverting "vendor/tinymce/tinymce/tinymce.min.js" to the old 7.11.15-version repairs it partly in DetailView, but that isn't a real solution.

#### Expected Behavior
The description-field should look as follows:
![email-good](https://user-images.githubusercontent.com/4035997/99068891-98ea6480-25ad-11eb-863f-e2b868d8361f.png)

See the way "Inhoud" (="description") field looks.
This is when clicking on the address in the Contacts ListView.

#### Actual Behavior
Currently the way this field looks when clicking on an e-mail address in the Contacts DetailView:
![email-wrong](https://user-images.githubusercontent.com/4035997/99069293-5e34fc00-25ae-11eb-9a1e-edb603b3e440.png)

Again, see the way "Inhoud" (="description") field looks. You can see that it's missing the wysiwyg-editor and goes out of bounds.

#### Steps to Reproduce

1. Enable using the itnernal e-mail client in your profile.
2. Click on an e-mail address in the Contacts ListView: this should work as expected.
3. Click on the same e-mail address of the same Contact in the Contacts DetailView: this shows the issue I'm experiencing.

#### Your Environment

* SuiteCRM Version used: 7.11.18
* Browser name and version (e.g. Chrome Version 51.0.2704.63 (64-bit)): Firefox and Chrome
* Environment name and version (e.g. MySQL, PHP 7): MariaDB 10.2, PHP 7.3
* Operating System and version (e.g Ubuntu 16.04):

**Comment by ebogaard** [comment]
Also, see #8913 for a related issue, which is only partly fixed, but closed.

**Comment by nelem** [comment]
Confirmed. TinyMCE is not included for ComposeVieweModal  but IS included for ComposeView. It also stops templates being included as they fail with a console error: unable to find tinymce editor.
It works OK in 7.11.5 but not in 7.11.18.

**Comment by nelem** [comment]
There is another installation of TinyMCE referenced by EmailUI at include/SugarTinyMCE.php. It may not be relevant here but it may be a ticking timebomb for later.

**Comment by ksaadDE** [comment]
Just as notification. We are currently experiencing the same Bug.

Maybe a warning BEFORE updating would be a good thing.

> I cleared the browser cache while testing this.
Reverting "vendor/tinymce/tinymce/tinymce.min.js" to the old 7.11.15-version repairs it partly in DetailView, but that isn't a real solution.

This "solution" didn't work (here). Somebody a better idea? ASAP would be great!

**Comment by nelem** [comment]
I spent many hours trying to fix this but it's beyond me. I rolled back to 7.11.15

**Comment by ksaadDE** [comment]
Maybe some change in the Code of TinyMCE? They did it a lot in the past. Such tiny changes.

**Comment by ksaadDE** [comment]
> I spent many hours trying to fix this but it's beyond me. I rolled back to 7.11.15

I see your effort, thanks for that! :100: :+1:  

Maybe that's not solvable currently.

**Comment by QuentinDoe** [comment]
rolled back to 7.11.15 for me too

**Comment by pstevens71** [comment]
So this does work.  The trick is flushing your browser cache while the email popup is open.  (and a couple of other steps).  Here's what I did:

1. Apply both fixes (I’m not sure which one did it) #8956 and 8913
2. Rebuilt all the JS files (in admin repair)
3. Click on the email address in the lead record to open up the email pop up. It will still not work. NOW… Shift F5 (in Chrome) to clear your browser cache with the email popup open… now works.

This has also been confirmed by another user in the forum using this procedure.

**Comment by QuentinDoe** [comment]
Bonjour , Nous vous confirmons la bonne réception de votre message par nos services .Nous mettons tout en oeuvre pour y répondre dans les meilleurs délais. A très vite, Vos dévoués courtiersCabinet l'assureur itinérant

**Comment by ellyqual** [comment]
faced the same issue here.
solution from pstevens71 did work, what I did:
1. apply fix from @mattlorimer [https://github.com/salesagility/SuiteCRM/commit/86a6654e170f98c902e14196d02a266b33de6343](url)
2. rebuilt all the JS files (in admin repair)
3. step 3 from @pstevens71 did not work for me, I had to clear my browser cache completely
4. tried to send an email via DetailView in Leads/Contact module: Popup will be shown correct and works fine & well even when creating mail from a template.

thank you guys!

**Comment by QuentinDoe** [comment]
Hello, it's good on my side this trick works

**Comment by davlet61** [comment]
Hello, I tried all the solutions above and none of them worked in neither 7.11.22 nor 7.12. 

I get this error: 

> TypeError: null is not an object (evaluating 'tinymce.activeEditor.setContent')

Any other suggestions

![Screenshot 2021-11-12 at 03 29 42](https://user-images.githubusercontent.com/30911739/141398719-0d004ff5-a3e8-4ac0-ad36-3c992cc64faa.png)
?

**Comment by elinazv** [comment]
Hi davlet61,
I've tried on versions 7.11.22 nor 7.12 (from git and download) and Email compose body worked for me from scratch:
Contact DetailView -> click on E-mail -> Email Compose

My Environment:
Browser name and version: Chrome Version 95.0.4638.69 on MacOs
Environment name and version : mysql 5.7, PHP 7.4
Docker: php:7.4-fpm-alpine3.13

Please provide more information about your environment.

**Comment by davlet61** [comment]
Hello elinazv, 
Thank you very much for your reply. 

It is a shared hosting with **PHP 7.4, mysql 5.6** and I have tried it on MacOS on Firefox, Chrome and Safari. 
Non of them had it working properly. I don't want to reinstall it and start over again.

**Comment by elinazv** [comment]
Hi davlet61,

now I've tested after the upgrade process and it also works for me.

So I did the following:
- I checked out your version: git checkout tags/v7.11.1
- At this moment I've our bug shown: Compose Email NOT  working properly
- After that, I've upgraded with package "7.12 Upgrade from 7.11.x"
- It was hard, I had to change some setting in my docker :-)
- But after a long process of upgrade, it really works fine

How do you upgrade? Please also check necessary permissions and make sure that you upgrade properly.

Actually, as I worked with SuiteCRM, we've simply checked out a new version having our customizations in other repository.

**Comment by davlet61** [comment]
Hi elinazv,

Thank you once again.

I upgraded it with Upgrade Wizard. 
All the necessary permissions are also set properly. 

So where are the customizations now?

**Comment by elinazv** [comment]
Hi davlet61,

here I mean the customizations as : custom fields, custom modules. So YOUR customizations to previous version.
But if you don't have any - so it is simpler.

As I guess from your message, Upgrade still didn't help? 

Deleting browser cache, Repair & Rebuild : All this you know, I suppose?

Actually, if you don't have any custom fields, modules, etc., why don't try to git checkout?

**Comment by davlet61** [comment]
Hi elinazv,

Sorry I misunderstood your sentence the first time:) 
I do have a plenty of customizations. (Fields, modules and whatnot) 

And nothing mentioned above helped me. I guess my final solution would be start over after all.

**Comment by elinazv** [comment]
Hi davlet61,

Yes, I suppose a solution for you is to start over as following:

- Backup your system with customizations
- Install a 7.11.22 clean version and make sure that you don't have a bug of Compose Email
- Apply step by step your customizations. I hope you don't have so much changes in "core" code :-)

Please let us know if it has worked for you.

**Comment by davlet61** [comment]
Hi elinazv,

So I did a fresh install and everything seemed to be working until I restored my db. 
I did not do any customisations, just uploaded the SQL file and it went back to the way it was.
Any suggestions?

Thank you!

**Comment by elinazv** [comment]
Hi davlet61,

this is a really interesting case - loading a database changes a design!
But you actually need a solution...

In fact, database by SuiteCrm is also represents a structure - modules, etc.

So as previously mentioned, I would also persuade a strategy of changes step by step. Eg. you load your database und you know what are customisations you did, so you can remove: 1. custom module, than 2. custom module and have a look, at what point it goes straight...

One more idea: Did you make much customisations by Contact? As the bug appears at Contact DetailView, it would be logical to suppose that some changes at Contact could by a reason for the bug...

As I remember, it is also possible to include custom files, eg. JavaScript files,  but as you say you didn't apply your code customisations, so it could not actually be a case...

It would be pleasure to help you!

**Comment by davlet61** [comment]
Hi elinazv,

Thank you so much for the support! 

The bug appears not only in Contact, but in Account and Lead as well. (Only in DetailView).

I am trying your solution right now.
If it doesn’t help, I will also take my time to maybe restore db table by table and see where it goes wrong. 

I will let you know the outcome once I’m finished with the process. 

Thank you!

**Comment by elinazv** [comment]
One more thing.
Be sure every time to delete cache.

Delete a cache folder directly in SuiteCRM.

Delete cache in browser.
I do it as following in Chrome:
F12 (you are sure aware of it, so you can see html, etc.) > If you press a button for refresh in Chrome some more time, you get 3 possibilities shown:
- Normal Reload
- Hard Reload
- Empty Cache and Hard Reload
So use every time the last!

And one thing more:
Don't use quick links at the left side (Recently Viewed) because it all is also cached.

Good luck!

**Comment by elinazv** [comment]
Hi davlet61,
please give a feedback, if this all has worked for you.
Best regards!

**Comment by davlet61** [comment]
Hi elinazv, 

Sorry for the late feedback. 

So I tried your suggestion, but it didn’t work when I removed custom tables. Also, if was kinda hard to follow and find all the other customizations within the db tables. So I did it backwards. 

I backed up my db and did a fresh install (7.12).
Then, I restored the tables one by one, just the ones I needed like Accounts, Contacts, Opportunities etc. Didn’t restore custom tables. Everything works as expected now. 
However I am still not sure what was causing that bug. I was too tired to try all the tables and see which one exactly is causing the bug:)

I imported the studio customizations from previous installation and reinstalled custom modules I was using and everything is fine.

Thank you so much for your support!
